src/main/java/cam72cam/immersiverailroading/model/part/DrivingAssembly.java

```diff
@@ -35,12 +39,25 @@ public static DrivingAssembly get(ValveGearConfig type, ComponentProvider provid
             return null;
         }
 
-        ValveGear left = ValveGear.get(wheels, type, provider, state, ModelPosition.LEFT.and(pos), 0);
-        ValveGear inner_left = ValveGear.get(wheels, type, provider, state, ModelPosition.INNER_LEFT.and(pos), 180);
-        ValveGear center = ValveGear.get(wheels, type, provider, state, ModelPosition.CENTER.and(pos), -120);
-        ValveGear inner_right = ValveGear.get(wheels, type, provider, state, ModelPosition.INNER_RIGHT.and(pos), 90);
-        ValveGear right = ValveGear.get(wheels, type, provider, state, ModelPosition.RIGHT.and(pos), center == null ? -90 : -240);
+        ValveGear left,
+                inner_left,
+                center,
+                inner_right,
+                right;
 
+        if(isLeftFirst) {
+            left = ValveGear.get(wheels, type, provider, state, ModelPosition.LEFT.and(pos), 0);
```

Comment:
Instead of copying and pasting this code, let's define a "direction offset" and multiply the angle by that.
```java
int angleDirection = isLeftFirst ? 1 : -1
...
right = ValveGear.get(wheels, type, provider, state, ModelPosition.RIGHT.and(pos), angleDirection * (center == null ? -90 : -240));
```

Reply:
Ok and I will change it later
Thanks